src/shared/components/ButtonBase/ButtonBase.style.ts

```diff
@@ -4,44 +4,75 @@ import styled from '@emotion/styled'
 
 import { colors, sizes, transitions } from '../../theme'
 
-export type ButtonVariant = 'primary' | 'secondary' | 'tertiary'
+export type ButtonVariant =
+  | 'primary'
+  | 'secondary'
+  | 'tertiary'
+  | 'destructive'
+  | 'destructive-secondary'
+  | 'warning'
+  | 'warning-secondary'
 export type ButtonSize = 'large' | 'medium' | 'small'
 export type ButtonBaseStyleProps = {
   variant: ButtonVariant
   size: ButtonSize
   clickable?: boolean
+  textOnly: boolean
+  iconOnly: boolean
 }
 
-const variantStyles = ({ variant }: ButtonBaseStyleProps): SerializedStyles => {
+const variantStyles = ({ variant, textOnly, iconOnly }: ButtonBaseStyleProps): SerializedStyles => {
   switch (variant) {
     case 'primary':
       return css`
         background-color: ${colors.blue[500]};
+        color: ${textOnly ? colors.blue[300] : colors.gray[50]};
+        border-color: ${colors.blue[300]};
+
+        path {
+          fill: ${textOnly && colors.blue[300]};
+        }
 
         &:hover {
           background-color: ${colors.blue[700]};
         }
 
         &:active {
           background-color: ${colors.blue[900]};
+          color: ${textOnly && colors.blue[400]};
+          border-color: ${colors.blue[400]};
+
+          path {
+            fill: ${textOnly && colors.blue[400]};
+          }
         }
       `
     case 'secondary':
       return css`
         /* 1px inset border */
         box-shadow: inset 0 0 0 1px ${colors.gray[500]};
+        border-color: ${colors.gray[50]};
+        color: ${colors.gray[50]};
 
         &:hover {
-          box-shadow: inset 0 0 0 1px ${colors.gray[50]};
+          box-shadow: inset 0 0 0 2px ${colors.gray[50]};
```

Comment:
```suggestion
          box-shadow: inset 0 0 0 2px ${colors.gray[300]};
          border-color: ${colors.gray[300]};
```
In the design colors are different for hover and active state. I'm a little confused why we are using box-shadow AND border. Maybe we should stick to just one of it(I think box-shadow would be preferable, because it doesn't change the size of button)